python/pyspark/sql/functions.py

```diff
@@ -363,6 +364,26 @@ def randn(seed=None):
     return Column(jc)
 
 
+@ignore_unicode_prefix
+@since(1.5)
+def sha1(col):
+    """
+    >>> from pyspark.sql import Row
+    >>> sqlContext.createDataFrame([Row(a='ABC')]).select(sha1('a').alias('hash')).collect()
+    [Row(hash=u'3c01bdbb26f358bab27f267924aa2c9a03fcfdb8')]
+
+    >>> sqlContext.createDataFrame([Row(a='ABC')]).select(sha1('a').alias('hash')).show()
```

Comment:
One tests case is enough for example (and coverage)

BTW, you could create a dataframe without `Row`:
```
 sqlContext.createDataFrame([('ABC',)], ['a']).select(sha1('a').alias('hash'))
```